src/daemons/StorageDaemon.cpp

```diff
@@ -109,6 +109,7 @@ int main(int argc, char *argv[]) {
                                                                       paths);
     if (!gStorageServer->start()) {
         LOG(ERROR) << "Storage server start failed";
+        gStorageServer->stop();
```

Comment:
Well done.  I want to know why the daemon crash?

Reply:
You can read the [documentation](https://en.cppreference.com/w/cpp/thread/thread/~thread) for the thread destructor.

> If *this has an associated thread (joinable() == true), std::terminate() is called.

Reply:
which thread?

Reply:
> nebula::WebService::wsThread_.

![image](https://user-images.githubusercontent.com/331435/69024415-9f2b1c80-09fd-11ea-97b0-23662ce6295a.png)

Reply:
I see.  Good catch. 
But i think a better way to solve the problem is to refactor web service.  It has lots of static members inside.  It is hard to handle the lifecycle.  Maybe we should remove them. WDYT?

Of course, we could open another pr for it.